morpheus/_lib/tests/llm/test_llm_node_runner.cpp

```diff
@@ -0,0 +1,119 @@
+#include <mrc/channel/forward.hpp>
+#include <mrc/coroutines/sync_wait.hpp>
+
+#include <coroutine>
+#include <memory>
+#include <stdexcept>
+#include <string>
+
+using namespace morpheus;
+using namespace morpheus::test;
+using namespace mrc;
+
+TEST_CLASS(LLMNodeRunner);
+
+auto make_nr_dummy_node()
+{
+    return llm::make_lambda_node([]() -> Task<int> {
+        co_return 0;
+    });
+}
+
+auto make_nr_single_input_node()
+{
+    return llm::make_lambda_node([](int i) -> Task<int> {
+        co_return i + 1;
+    });
+}
+
+TEST_F(TestLLMNodeRunner, ParentInput)
+{
+    auto node   = make_nr_single_input_node();
+    auto inputs = llm::input_mappings_t{{"input0", "arg0"}};
+    llm::LLMNodeRunner runner{"runner", inputs, node};
+    ASSERT_EQ(runner.parent_input_names(), std::vector<std::string>{"input0"});
+    ASSERT_EQ(runner.sibling_input_names(), std::vector<std::string>{});
+}
+
+TEST_F(TestLLMNodeRunner, SiblingInput)
+{
+    auto node_1   = make_nr_single_input_node();
+    auto inputs_1 = llm::input_mappings_t{{"/ext", "arg0"}};
+    llm::LLMNodeRunner runner_1{"runner", inputs_1, node_1};
+    ASSERT_EQ(runner_1.parent_input_names(), std::vector<std::string>{});
+    ASSERT_EQ(runner_1.sibling_input_names(), std::vector<std::string>{"/ext"});
+
+    auto node_2   = make_nr_single_input_node();
+    auto inputs_2 = llm::input_mappings_t{{"/ext/input0", "arg0"}};
+    llm::LLMNodeRunner runner_2{"runner", inputs_2, node_2};
+    ASSERT_EQ(runner_2.parent_input_names(), std::vector<std::string>{});
+    ASSERT_EQ(runner_2.sibling_input_names(), std::vector<std::string>{"/ext/input0"});
+}
+
+TEST_F(TestLLMNodeRunner, DuplicateInput)
+{
+    auto node   = make_nr_single_input_node();
+    auto inputs = llm::input_mappings_t{{"input0", "arg0"}, {"input0", "arg0"}};
+    ASSERT_THROW(llm::LLMNodeRunner("runner", inputs, node), std::runtime_error);
+}
+
+TEST_F(TestLLMNodeRunner, InvalidExternalNode)
+{
+    auto node   = make_nr_single_input_node();
+    auto inputs = llm::input_mappings_t{{"ext/input0", "arg0"}};
+    ASSERT_THROW(llm::LLMNodeRunner("runner", inputs, node), std::invalid_argument);
+}
+
+TEST_F(TestLLMNodeRunner, InvalidInputForNode)
+{
+    auto node     = make_nr_single_input_node();
+    auto inputs_1 = llm::input_mappings_t{{"input0", "arg1"}};
+    ASSERT_THROW(llm::LLMNodeRunner("runner", inputs_1, node), std::runtime_error);
+
+    auto inputs_2 = llm::input_mappings_t{{"input0", "arg0"}, {"input1", "arg1"}};
+    ASSERT_THROW(llm::LLMNodeRunner("runner", inputs_2, node), std::runtime_error);
+}
+
+TEST_F(TestLLMNodeRunner, Execute)
+{
+    llm::LLMNode node;
+
+    auto runner_1 = node.add_node("Root1", {}, make_nr_dummy_node());
+    auto runner_2 = node.add_node("Root2", {{"/Root1"}}, make_nr_single_input_node(), true);
+
+    auto context = std::make_shared<llm::LLMContext>(llm::LLMTask{}, nullptr);
+
+    coroutines::sync_wait(runner_1->execute(context));
+    coroutines::sync_wait(runner_2->execute(context));
```

Comment:
Important: run the executions in parallel.
```suggestion
    coroutines::sync_wait(
        coroutines::when_all(
            runner_1->execute(context),
            runner_2->execute(context)
        )
    );
```